clang/lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -722,8 +735,14 @@ void CudaToolChain::addClangTargetOptions(
       llvm::sys::path::append(WithInstallPath, Twine("../../../share/clc"));
       LibraryPaths.emplace_back(WithInstallPath.c_str());
 
+      // Select remangled libclc variant. 64-bit longs default, 32-bit longs on
+      // Windows
       std::string LibSpirvTargetName =
           "remangled-l64-signed_char.libspirv-nvptx64--nvidiacl.bc";
+      if (HostTC.getTriple().isOSWindows())
+        LibSpirvTargetName =
+            "remangled-l32-signed_char.libspirv-nvptx64--nvidiacl.bc";
```

Comment:
There doesn't seem to be any tests checking that the proper `.bc` file is being picked up.  Is it necessary?

Reply:
That is a good idea. Ill write a test to check that the correct file is being used.

Reply:
Added test to check .bc file: clang/test/Driver/cuda-nvptx-target.cpp